### ISSUE TITLE
libsodium vs sodium

### ISSUE THREAD

**Issue by SvenAlHamad** [paragonie/halite#61]
Hi,

I've followed this guide to install the php extension: https://paragonie.com/book/pecl-libsodium/read/00-intro.md#installing-extension

This installed an extension called sodium, not libsodium (ref: https://github.com/jedisct1/libsodium-php)

While when trying to add paragonie/halite via composer, the command failed with an error:
" paragonie/halite dev-master requires ext-libsodium ^1.0.6 -> the requested PHP extension libsodium is missing from your system."

Checking the composer.json file in master reveals that the old libsodium (v1.x) is still required, and installing the latest "sodium" lib doesn't work.
```
    "require": {
        "php": "^7",
        "paragonie/constant_time_encoding": "^2",
        "ext-libsodium": "^1.0.6"
    },
```

**note**: This might be linked to an existing issue: https://github.com/paragonie/halite/issues/51

**Comment by paragonie-scott** [comment]
It's unrelated to #51.

What happened was, ext-libsodium 1.0.6 was the last before sodium 2.0.0 was released. This happened because the RFC vote ended up in a stupid place. (See second vote.) https://wiki.php.net/rfc/libsodium#proposed_voting_choices

I'll push a minor update to all supported versions of Halite to use the new sodium_* API when I get the chance, but I'm busy with paid client work at the moment.

**Comment by SvenAlHamad** [comment]
Thanks,

I'll probably install libsodium 1.0.6 and remove the v2 I currently have, that should sort me out for time being, until the repo gets updated.

Feel free to close the ticket if you want, or you can leave it for others as a reference.

Cheers

**Comment by paragonie-scott** [comment]
This will be fixed in Halite v3.3
